fly/commands/pipelines.go

```diff
@@ -28,19 +28,26 @@ func (command *PipelinesCommand) Execute([]string) error {
 	}
 
 	var headers []string
-	var pipelines []atc.Pipeline
+	var unfilteredPipelines []atc.Pipeline
 
 	if command.All {
-		pipelines, err = target.Client().ListPipelines()
+		unfilteredPipelines, err = target.Client().ListPipelines()
 		headers = []string{"name", "team", "paused", "public", "last updated"}
 	} else {
-		pipelines, err = target.Team().ListPipelines()
+		unfilteredPipelines, err = target.Team().ListPipelines()
 		headers = []string{"name", "paused", "public", "last updated"}
 	}
 	if err != nil {
 		return err
 	}
 
+	var pipelines []atc.Pipeline
+	for _, p := range unfilteredPipelines {
```

Comment:
It's interesting how this creates a difference between the json that comes in from the API and the json that's displayed through `displayhelpers.JsonPrint`:

```console
$ fly -t local pipelines --json
null
```

vs

```console
$ fly -t local pipelines --verbose

GET /api/v1/teams/main/pipelines HTTP/1.1
HTTP/1.1 200 OK <...>
[
  {
    "id": 1,
    "name": "test",
    "paused": true,
    "public": false,
    "archived": true,
    "team_name": "main",
    "last_updated": 1585749902
  }
]
```

while I do understand that the purpose of `--json` is to literally just change the way that the information we want to show is displayed (instead of passing the response through), but it made me wonder if we should use a query parameter to limit the API's response in the first place :thinking:  have you thought about that? not sure how that'd affect future plans for the dashboard/fly. 

thanks!

Reply:
It did feel weird to me making `--json` differ from the API response, but it felt even weirder to get different data when specifying `--json` compared to the table view. If you have differing thoughts there I'd be curious to hear them - @jomsie and I were struggling a bit with getting the UX to feel right.

In terms of pagination, that definitely is something we've considered. @pivotal-jamie-klassen added a "tech debt" note [here](https://project.concourse-ci.org/projects/MDc6UHJvamVjdDM4Mzc3ODI=) to paginate all the "dashboard" APIs. I feel it would probably need to become its own epic since it'll likely require a fair bit of work to maintain feature parity with what we have now (e.g. filtering the dashboard becomes more difficult). It would be nice for the dashboard to refresh only the 200 jobs that are visible every 5 seconds, rather than the 30,000 jobs in the entire cluster.

Reply:
>  If you have differing thoughts there I'd be curious to hear them 

tbh, I think it's great as is right now - I've always seen `json` as just a different view of the default `table`, which I think makes sense to have the behavior of "show the very same data but in a different way"

> In terms of pagination,

oh, sorry, I was actually referring to something like `?archived=false` (so that `archived:true` pipelines wouldn't even reach the client) or something like that (more like filtering, rather than paginating)

*(even though paginating would be great too!)*

Reply:
Oh haha sorry, I misunderstood.

The [RFC](https://github.com/concourse/rfcs/pull/33/files#diff-36359b59bfd438f0b68f9be6fefac7d0R28) specifies:

> `ListAllPipelines`, `ListPipelines`, and `GetPipeline` will continue to return archived pipelines. Hiding archived pipelines is the job of the web UI and `fly`, not the API.

Now, that's not to say having a query parameter on `ListAllPipelines` and `ListPipelines` wouldn't also make sense...

Reply:
PS I fixed `--json` to output `[]` when there are no non-archived pipelines rather than `null`